Automattic/wp-calypso
# Notifications - Resolve inconsistent behavior due to incorrect and stale isShowing values.

Related to # https://github.com/Automattic/wp-calypso/issues/88344 

## Proposed Changes

Fixes an above issue where notifications were being marked as seen even though the panel had never been opened. Also resolves various other issues that were present and blocking this code and initial fix from working correctly.

* Changes the `isShowing` prop to initialize as `false`. Starting as `true` was forcing notes to think they were open when mounting and thus marking notes as being seen when they were not.
* Reorganizes the standalone app to use a `<NoteWrapper />` component that stores `isShowing` and `isVisible` in component state, forcing the notification component to update with these props. The previous method of storing state for `isShowing` etc. as a local variable is problematic as changing this value never causes the Notifications component to update and get the new prop value. 
* Changes the refresh handler that is defined on mount to use `this.props.isShowing` as opposed to the destructured value. The destructured value will never change, while referencing props directly will give us an accurate value.
* Fixes `showing` and `visible` properties in the debug call `updateLastSeenTime 1` to pass the actual values (there was a typo causing these to always be undefined).
* Adds a handful of other debug calls that I found necessary for this. Since the dev workflow is not in a great state, I think it would be good to keep these debug calls in place to make any further issues easier to diagnose.

Before this change the standalone component was initializing the wrong state, not causing notifications to update on state changes, and then always refreshing based off the initial mounting value of `isShowing`. Since the initial value was `true`, this made the panel essentially always think it was open...

## Testing Instructions



* Update this version of notifications on your sandbox by running the `install-plugin.sh notifications try/fix-notificaitons-falsely-seen` command. NOTE - the dev build (`yarn dev --sync`) from calypso directly does not work correctly.
* Sandbox widgets AND a simple test site. DO NOT sandbox public-api.
* Close all other tabs with a notification bell or site preview.
* Visit the homepage of your test site. 
* If you do not have an orange dot, trigger a notification for yourself. You can probably do this by commenting on one of your own posts in an incognito window. I have been using wp cli commands on my sandbox `wp comment create --comment_post_ID=$postID --comment_author="somename" --url=$siteURL --comment_content="some unique content"`. note content will have to change every time as duplicates are failures to prevent spam.
* Without opening the notifications panel, reload the browser window and let it reload. Do this a couple times. The notification dot should not disappear and there should be no request to `/notifications/seen` being sent.
* Open the panel, verify notes work as expected.
* Reload the page, verify the orange dot is no longer present.
* If you would like to see the debug commands, you can set `localStorage.setItem( 'debug', 'notifications:*' )` in the console and reload to see all notifications related debug logs.

Before this change, a request to `/notifications/seen` would go out without opening the panel and the orange dot would disappear after a couple reloads. This should no longer happen.

## Pre-merge Checklist
